### ISSUE TITLE
is there any way to get subject_hash_old?

### ISSUE THREAD

**Issue by doorholez** [nimpo/hash-bash-ssl#1]
I am trying to do the `subject_hash_old` manually and i've learned a lot from your repo. Someone said the only difference is the hash algorithm, but i changed sha1sum to md5sum in the script and get different result with openssl. There should be some other differences that i don't know. However, i'm not good at C so I'm not able to understand what happened in the source code of openssl:(
Do you know how `subject_hash_old` works? Could you please kindly teach me that? Thanks a lot!

**Comment by nimpo** [comment]
Yes, there are other subtleties around canonicalization, IIRC . Off the top of my head I can't remember but I will have a think.

**Comment by doorholez** [comment]
Hi nimpo
Thanks a lot for your kindness! I'm sorry to bother you further but I ran the new script and got different result with openssl. (The result for new hash is the same but different for old hash) Maybe there are some mistakes in the new sctipt for old hash. Could you please correct that?

**Comment by nimpo** [comment]
Hi doorholez,
Ha, yes! I just picked a random certificate that I had to hand as my test case and didn't see my error. Its subject name is just a /cn=blah. The mistake, I think, is wrapping every RDN set with a sequence, when I should just have wrapped the whole group of sets in a sequence. The old hash algorithm just takes the whole DN dumped from the certificate and runs the md5sum on it; the new hash takes the DN, shaves off the SEQUENCE that wraps it, canonicalizes it (spaces; and stringtypes) and takes the sha1sum.
